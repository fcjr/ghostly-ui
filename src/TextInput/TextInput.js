import React from "react";
import "./TextInput.scss";

const TextInput = ({
  type = "text",
  label,
  placeholder,
  value,
  onChange,
  helpText
}) => (
  <div className="TextInput__formGroup">
    {label && <label className="TextInput__label">{label}</label>}
    <input
      type={type}
      className="TextInput__input"
      value={value}
      onChange={e => onChange && onChange(e.target.value)}
      placeholder={placeholder}
    />
    {helpText && <div className="TextInput__helpText">{helpText}</div>}
  </div>
);

export default TextInput;
