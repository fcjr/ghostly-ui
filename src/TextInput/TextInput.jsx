import React from 'react';
import style from './TextInput.module.scss';

const TextInput = ({
	type = 'text',
	label,
	placeholder,
	value,
	onChange,
	helpText
}) => (
	<div>
		{label && <label>{label}</label>}
		<input
			type={type}
			className={style.input}
			value={value}
			onChange={(e) => onChange && onChange(e.target.value)}
			placeholder={placeholder}
		/>
		{helpText && <div>{helpText}</div>}
	</div>
);

export default TextInput;
