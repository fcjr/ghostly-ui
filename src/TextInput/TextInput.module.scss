.input {
  background-color: #f3f3f3;
  border: 1px solid #e1e1e1;
  border-radius: 4px;
  padding: 0 10px;
  height: 30px;

  &:focus {
    outline: none;
    border: 1px solid #720174;
  }
}
