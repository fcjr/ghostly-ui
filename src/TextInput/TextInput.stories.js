import React from 'react';

import TextInput from './TextInput';

export default {
  title: 'Inputs/FormInputs/TextInput',
  component: TextInput,
};

const Template = (args) => <TextInput {...args} />;

export const Email = Template.bind({});
Email.args = {
    label: "Email",
    placeholder: "name@email.com",
};

export const EmailPopulated = Template.bind({});
EmailPopulated.args = {
    label: "Email",
    placeholder: "name@email.com",
    value: "email@example.com"
};
