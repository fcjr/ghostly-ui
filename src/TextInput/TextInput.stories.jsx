import React from 'react';

import TextInput from './TextInput';

export default {
	title: 'Inputs/TextInput',
	component: TextInput,
	argTypes: {
		label: { control: 'text' },
		placeholder: { control: 'text' },
		value: { control: 'text' },
		errorText: { control: 'text' },
		onChange: { action: 'changed' }
	},
};

const Template = (args) => <TextInput {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const WithLabel = Template.bind({});
WithLabel.args = {
	label: 'email',
	placeholder: 'name@example.com',
};

export const Error = Template.bind({});
Error.args = {
	label: 'email',
	placeholder: 'name@example.com',
	errorText: 'Invalid Email',
};
