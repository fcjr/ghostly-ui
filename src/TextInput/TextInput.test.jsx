import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from './TextInput';

describe('TextInput', () => {
	it('renders properly', () => {
		const tree = renderer
			.create(<TextInput label="Email" placeholder="name@example.com" onChange={() => {}} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
