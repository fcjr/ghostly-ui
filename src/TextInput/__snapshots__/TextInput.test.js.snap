// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders properly 1`] = `
<div
  className="TextInput__formGroup"
>
  <label
    className="TextInput__label"
  >
    Email
  </label>
  <input
    className="TextInput__input"
    onChange={[Function]}
    placeholder="name@example.com"
    type="text"
  />
</div>
`;
